rllib/models/action_dist_v2.py

```diff
@@ -0,0 +1,119 @@
+from typing import Tuple
+import gym
+import abc
+
+from ray.rllib.utils.annotations import ExperimentalAPI
+from ray.rllib.utils.typing import TensorType, Union, ModelConfigDict
+
+
+@ExperimentalAPI
+class ActionDistributionV2(abc.ABC):
+    """The policy action distribution of an agent.
+
+    Args:
+        inputs: input vector to define the distribution over.
+
+    Examples:
+        >>> model = ... # a model that outputs a vector of logits
+        >>> action_logits = model.forward(obs)
+        >>> action_dist = ActionDistribution(action_logits)
+        >>> action = action_dist.sample()
+        >>> logp = action_dist.logp(action)
+        >>> kl = action_dist.kl(action_dist2)
+        >>> entropy = action_dist.entropy()
+
+    """
+
+    @abc.abstractmethod
+    def sample(
+        self,
+        *,
+        sample_shape: Tuple[int, ...] = None,
+        return_logp: bool = False,
+        **kwargs
+    ) -> Union[TensorType, Tuple[TensorType, TensorType]]:
+        """Draw a sample from the action distribution.
+
+        Args:
+            sample_shape: The shape of the sample to draw.
```

Comment:
Can we be more specific here? Doesn't the input (currently `self.inputs`) already determine the batch/time dimensions and the rest is "fixed"? If this is not the case, can we add examples to this docstring?

Reply:
This should be addressed in the docstrings of each individual distribution.